router_v1/GluexRouter.sol

```diff
@@ -217,39 +209,50 @@ contract GluexRouter is EthReceiver {
             finalOutputAmount = outputBalanceAfter - outputBalanceBefore;
         }
 
+        // Surplus calculation
+        if (finalOutputAmount >= desc.outputAmount && desc.outputAmount > desc.effectiveOutputAmount) {
+            surplus = desc.outputAmount - desc.effectiveOutputAmount;
+            finalOutputAmount = finalOutputAmount - surplus;
+        }  else if (desc.outputAmount > finalOutputAmount && finalOutputAmount > desc.effectiveOutputAmount) {
+            surplus = finalOutputAmount - desc.effectiveOutputAmount;
+            finalOutputAmount = finalOutputAmount - surplus;
+        } else {
+            surplus = 0;
+            finalOutputAmount = finalOutputAmount;
+        }
+
         // Ensure final output amount meets the minimum required
         if (finalOutputAmount < desc.minOutputAmount) revert("Negative slippage too large");
 
-        uint256 surplus = 0;
-
-        if (desc.externalOutputAmount > 0 && finalOutputAmount > desc.externalOutputAmount) {
-            // Part of surplus is shared with the partners and treasury
-            surplus = finalOutputAmount - desc.externalOutputAmount;
-            uint256 partnerShare = 0;
-            uint256 routingShare = 0;
-
-            // Transfer partner share to the partner address
-            if (desc.partnerAddress != address(0)) {
-                partnerShare = (surplus * desc.partnerSurplus) / 10000;
-                if (partnerShare > 0) {
-                    uniTransfer(IERC20(desc.outputToken), desc.partnerAddress, partnerShare);
-                }
+        if (surplus > 0) {
+            // Calculate and transfer partner surplus
+            uint256 partnerShare = (surplus * desc.partnerSurplusShare) / 10000;
+            if (partnerShare > 0) {
+                uniTransfer(
+                    IERC20(desc.outputToken),
+                    desc.partnerAddress,
+                    partnerShare
+                );
             }
 
-            // Transfer routing share to the treasury address
-            routingShare = (surplus * desc.routingSurplus) / 10000;
+            // Calculate and transfer routing surplus
+            uint256 routingShare = surplus - partnerShare;
             if (routingShare > 0) {
```

Comment:
why is this here? I can't think of a case in which protocol share would be negative or equal to zero